contributed_definitions/nyaml/NXpiezo_config_spm.yaml

```diff
@@ -0,0 +1,192 @@
+category: base
+doc: |
+  A base class describing piezo settings for scanning probe microscopy.
+  
+  The piezoelectric material of SPM actuators gets deformed due to the applied electric field.
+  Description below shows calibration coefficients and other configuration parameters of the piezo actuators.
+
+# Discussion: No need to create this base class rather we can define in spm application definition.
+type: group
+NXpiezo_config_spm(NXobject):
+  piezo_material(NXpiezoelectric_material):
+    doc: |
+      The material description and properties of the piezoelectric scanner materials.
+    curvature_radius_N(NX_NUMBER):
+      unit: NX_LENGTH
+      doc: |
+        The N (substring) denotes X or Y. There are 2 parameters in X and Y directions. It can be set
+        approximately to the length of the piezo tube along X and Y axis.
+  calibration(NXcalibration):
+    calibration_type(NX_CHAR):
+      doc: |
+        The name of the calibration type, sometimes it is called
+        `active calibration`.
+      enumeration: [active, passive]
+    calibration_name:
+      doc: |
+        A specific name of the calibration (e.g. active type with name 'LHe').
+    caliberation_date(NX_DATE_TIME):
```

Comment:
`calibration_date`.

Please ensure that any specialized names fit those in the `NXcalibration` base class.

Reply:
Typo: noted to be implemented!